### PR TITLE
Defining basic authentication

### PR DESCRIPTION
**Pull request by danielfigueiredo**
Supported methods are local (cash counter), facebook, google and github
I also added bluebird to our project and dropped observables, we can brign observables back when we actually need streaming and continuous events, like using socket.io/etc
authorization is not implemented, and will be my next task
@mattmcfad I know you've been developing authorization for another project you're working, let me know if you want to be part of the discussion for schema design
@tyllrb this can be a small schema meeting just to talk about roles/permissions/user data model, not the schema for the entire app
let me know if you guys want to join

### CODE REVIEW

File: app/apis/database-api.js [hunk 1]
```diff
@@ -1,41 +1,37 @@
 'use strict';
 
+const configResource = require('../services/config-service');
+const BluePromise = require('bluebird');
```
Comment (mattmcfad) [inline app/apis/database-api.js]: Could rename to `Promise` for simplicity.

File: app/app.js [hunk 2]
```diff
@@ -23,21 +44,17 @@ const server = http.createServer(app);
...
-      server.listen(app.get('port'), () => {
-        observer.next(app);
-        observer.complete();
-      });
-    }, (err) => {
-      observer.error(err);
-    });
-  });
+  return databaseAPI.connect().then(() =>
+    BluePromise.promisify(server.listen, {context: server})(app.get('port'))
```
Comment (mattmcfad) [inline app/app.js:48]: 💪

File: app/configs/dev.json [hunk 3]
```diff
@@ -3,5 +3,14 @@
...
+  "NODE_TEST_PORT": 3011,
+  "APP_SESSION_NAME": "CashCounter",
+  "APP_SESSION_SECRET": "thereisnodeath",
+  "SERVER_URL": "http://localhost:3010",
+  "GOOGLE_CLIENT_ID": "558974727815-61jq51p1pp7v40lacsf7l54idvhop450.apps.googleusercontent.com",
+  "GOOGLE_CLIENT_SECRET": "jN7lnCPAz4H8l3RiUrh1NmFk",
+  "FACEBOOK_CLIENT_ID": "1679462882305886",
+  "FACEBOOK_CLIENT_SECRET": "e43a307f0cfa92857ddc6b9aeb18b5d5",
+  "GIT_HUB_CLIENT_ID": "e6d65ae2fea1a53424a1",
+  "GIT_HUB_CLIENT_SECRET": "a6a59912b5cb4d404769b6a881ca32ec499f01cf"
```
Comment (mattmcfad) [inline app/configs/dev.json:15]: eh... should we move to a `.private/` config file?

File: app/providers/common-provider.js [hunk 4]
```diff
@@ -1,66 +1,26 @@
 'use strict';
 
-const mongoose = require('mongoose');
-const Rx = require('rxjs');
+const BluePromise = require('bluebird');
+const mongoose = BluePromise.promisifyAll(require('mongoose'));
```
Comment (mattmcfad) [inline app/providers/common-provider.js:4]: 💪

File: app/configs/dev.json (hunk 3 above)
Comment (danielfigueiredo) [inline app/configs/dev.json:15]: what you mean? create another file for keys and secrets?
I thought this was already it =P

**Comment by danielfigueiredo** [comment]
So I'm declaring it has `BluePromise` because now in Nodejs `Promise` is global prototype we can use, thus if I use `Promise` I would be shadowing the original `Promise` and that might be confusing.
Moreover I read that promises from blue bird are faster !

**Comment by mattmcfad** [comment]
I am a fan of promises over observables. Observables are still foreign to me but I am open to bringing it back for when they needed.

@danielfigueiredo I'm interested in participating in the schema design meeting for authorization - let me know a time other than Wednesday or Saturday.

LGTM :shipit:

**Comment by mattmcfad** [comment]
@danielfigueiredo thanks for clarifying - `BluePromise` it is :)

File: app/configs/dev.json (hunk 3 above)
Comment (mattmcfad) [inline app/configs/dev.json:15]: Ah, I guess I meant have this file uncommitted. 

If you're okay with these secrets being public then carry on.

Comment (danielfigueiredo) [inline app/configs/dev.json:15]: you have a point! I created this org and the things with cash counter account, so gmail and facebook are all for it
unfortunately we have to pay for it to be private
let's pray for people to not get it hahah

**Pull request merged**